partition/src/apex.rs

```diff
@@ -152,11 +152,12 @@ impl ApexSamplingPortP4 for ApexLinuxPartition {
         sampling_port_id: SamplingPortId,
         message: &mut [ApexByte],
     ) -> Result<(Validity, MessageSize), ErrorReturnCode> {
-        if let Some((port, val)) = SAMPLING_PORTS
-            .read()
-            .unwrap()
-            .get(sampling_port_id as usize - 1)
-        {
+        let read = if let Ok(read) = SAMPLING_PORTS.read() {
+            read
+        } else {
+            return Err(ErrorReturnCode::NotAvailable);
+        };
+        if let Some((port, val)) = read.get(sampling_port_id as usize - 1) {
```

Comment:
Can `sampling_port_id` be zero? I see potential for an unsigned integer wrap here. If not, I would comment it at least.

Reply:
The `sampling_port_id` is always greater than zero, since it is (incidentally) derived from the current length, after inserting an element, of the `ArrayVec` storing the sampling ports. https://github.com/aeronautical-informatics/apex-linux/pull/15/files#diff-d21de3347b5e6174c93c455b533568b798d139a47f7ba2564655c0731e27f889R120